harness/determined/pytorch/_pytorch_context.py

```diff
@@ -659,7 +667,7 @@ def clip_grads(params):
         # before we apply gradient clipping and `step()`. In the case of APEX
         # this is called in backward() instead, so that it's inside the context
         # manager and before unscaling.
-        if self.distributed.size > 1 and not self._use_apex:
+        if self.distributed.size > 1 and self._distributed_backend.use_horovod() and not self._use_apex:
```

Comment:
IIUC, this is ok because DDP synchronizes gradients during loss.backward().  If that's true, this is worth another comment.